packages/shared/topology/sidebar/deployments/DeploymentResources.tsx

```diff
@@ -0,0 +1,166 @@
+};
+
+type DeploymentResourceProps = {
+  deployment: DeploymentKind;
+};
+
+const DeploymentResources: React.FC<DeploymentResourceProps> = ({
+  deployment,
+}) => {
+  const { t } = useCustomTranslation();
+  const [pods, podsLoaded, podsLoadError] = useK8sWatchResource<
+    PodWithMetricsKind[]
+  >({
+    kind: PodModel.kind,
+    isList: true,
+    namespaced: true,
+    namespace: CEPH_STORAGE_NAMESPACE,
+  });
+
+  const [statefulSets, statefulSetLoaded, statefulSetError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfStatefulSetResource);
+
+  const [replicaSets, replicaSetsLoaded, replicaSetsError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfReplicaSetResource);
+
+  const [daemonSets, daemonSetsLoaded, daemonSetError] =
+    useK8sWatchResource<K8sResourceCommon[]>(odfDaemonSetResource);
+
+  const memoizedStatefulSets = useDeepCompareMemoize(statefulSets, true);
+  const memoizedReplicaSets = useDeepCompareMemoize(replicaSets, true);
+  const memoizedDaemonSets = useDeepCompareMemoize(daemonSets, true);
+
+  const filteredPods = React.useMemo(() => {
+    const resources = [
+      ...memoizedReplicaSets,
+      ...memoizedDaemonSets,
+      ...memoizedStatefulSets,
+      deployment,
+    ];
+    const loaded =
+      statefulSetLoaded && replicaSetsLoaded && daemonSetsLoaded && podsLoaded;
+    const loadError =
+      statefulSetError || replicaSetsError || daemonSetError || podsLoadError;
+    return loaded && !loadError && !_.isEmpty(pods)
+      ? pods.filter((pod) => isPodInDeployment(pod, ...resources))
+      : [];
+  }, [
+    daemonSetError,
+    daemonSetsLoaded,
+    deployment,
+    memoizedDaemonSets,
+    memoizedReplicaSets,
+    memoizedStatefulSets,
+    pods,
+    podsLoadError,
+    podsLoaded,
+    replicaSetsError,
+    replicaSetsLoaded,
+    statefulSetError,
+    statefulSetLoaded,
+  ]);
+
+  const basePath = usePrometheusBasePath();
+  const [usedCpu, errorUsedCpu, loadingUsedCpu] = useCustomPrometheusPoll({
+    endpoint: 'api/v1/query' as any,
+    query: POD_QUERIES[PodMetrics.CPU],
+    basePath: basePath,
+  });
+  const [usedMemory, errorUsedMemory, loadingUsedMemory] =
+    useCustomPrometheusPoll({
+      endpoint: 'api/v1/query' as any,
+      query: POD_QUERIES[PodMetrics.MEMORY],
+      basePath: basePath,
+    });
+
+  for (let podIndex = 0; podIndex < filteredPods.length; podIndex++) {
+    filteredPods[podIndex].metrics = { cpu: NaN, memory: NaN };
+
+    if (usedCpu && _.isEmpty(errorUsedCpu) && !loadingUsedCpu) {
+      filteredPods[podIndex].metrics.cpu = getPodMetric(
+        usedCpu.data.result,
+        filteredPods[podIndex].metadata.name
+      );
+    }
+    if (usedMemory && _.isEmpty(errorUsedMemory) && !loadingUsedMemory) {
+      filteredPods[podIndex].metrics.memory = getPodMetric(
+        usedMemory.data.result,
+        filteredPods[podIndex].metadata.name
+      );
+    }
+  }
+
+  return (
+    <div className="co-m-pane__body topology-sidebar-tab-observe">
```

Comment:
As I forgot to update this in my previous PR, I can take care of it in my StorageCluster tabs PR.
```suggestion
    <div className="odf-m-pane__body topology-sidebar-tab-observe">
```